tools/dwp_tools.py

```diff
@@ -41,13 +42,95 @@ def flip_lines(collection):
+    
+    records = [] 
+    
+    for feature in line_col:
+        line_id = feature['properties'].get('line_id', None)
+        name = feature['properties'].get('name', None)
+        
+        if type(feature['geometry']['coordinates'][0][0]) != tuple:
+            line = TLine(feature['geometry']['coordinates'])
+        else:
+            line = TMultiLineString(feature['geometry']['coordinates'])     
+    
+        legger_col = {
+            'waterpeil': feature['properties'].get('waterpeil', 0),
+            'waterdiepte': feature['properties'].get('waterdiepte', 0),
+            'breedte_wa': feature['properties'].get('breedte_wa', 0),
+            'bodemhoogte': feature['properties'].get('bodemhoogte', 0),
+            'bodembreedte': feature['properties'].get('bodembreedte', 0),
+            'talud_l': feature['properties'].get('talud_l', 0),
+            'talud_r': feature['properties'].get('talud_r', 0),
+            'peiljaar': feature['properties'].get('peiljaar', 0)
+        }
+
+        # Berekenen leggerwaarden
+        ti_velden_col = vul_leggerwaarden(legger_col)
+    
+        # Berekenen locatie leggerpunten
+        profiel_dict = create_leggerpunten(line, line_id, name, ti_velden_col['ti_waterbr'],
+                                           ti_velden_col['ti_talulbr'], ti_velden_col['ti_knkbodr'])
+    
+        # Berekenen waarden leggerpunten
+        legger_point_dict = update_leggerpunten_diepten(profiel_dict, ti_velden_col)
+        legger_point_dict['peiljaar'] = legger_col['peiljaar']
+        
+        legger_point_dict_22L = legger_point_dict.copy()
+        legger_point_dict_22L['puntcode'] = 22
+        legger_point_dict_22L['volgnr'] = 1
+        legger_point_dict_22L['afstand'] = 0
+        legger_point_dict_22L['z_waarde'] = legger_point_dict_22L['L22_peil']
+        
+        legger_point_dict_99L = legger_point_dict.copy()
+        legger_point_dict_99L['puntcode'] = 99
+        legger_point_dict_99L['volgnr'] = 2
+        legger_point_dict_99L['afstand'] = legger_point_dict_99L['ti_talulbr']
+        legger_point_dict_99L['z_waarde'] = legger_point_dict_99L['knik_l_dpt']
+                
+        legger_point_dict_99R = legger_point_dict.copy()
+        legger_point_dict_99R['puntcode'] = 99
+        legger_point_dict_99R['volgnr'] = 3
+        legger_point_dict_99R['afstand'] = legger_point_dict_99R['ti_knkbodr']        
+        legger_point_dict_99R['z_waarde'] = legger_point_dict_99R['knik_r_dpt']
+                
+        legger_point_dict_22R = legger_point_dict.copy()
+        legger_point_dict_22R['puntcode'] = 22
+        legger_point_dict_22R['volgnr'] = 4
+        legger_point_dict_22R['afstand'] = legger_point_dict_22R['ti_waterbr']        
+        legger_point_dict_22R['z_waarde'] = legger_point_dict_22R['R22_peil']
+            
+        # Vullen point collection voor deze lijn
+        legger_point_col = MemCollection(geometry_type='Point')       
```

Comment:
Ah ja, nog restant van toen in (onterecht) de writerecords in de for loop had zitten... de write er wel uit gehaald, maar het aanmaken van de collection vergeten ;)